### DISCUSSION TITLE
Unmarshal type

### DISCUSSION THREAD

**Discussion by Bzelga** [apache/camel-karavan#1184] [Q&A] [answered]
Hi.
I'm trying to do a simple integration where there will be an xml to POJO conversion.
I can't figure out how to fill in the unmarshal type correctly.
I found an example that I tried to reproduce: https://stackoverflow.com/questions/76234573/unmarshalling-xml-json-to-class-in-apache-karavan
Created the "User" class and the following route:
```
- route:
    id: route-6cc2
    nodePrefixId: route-9ba
    from:
      id: from-15db
      uri: kamelet:timer-source
      parameters:
        message: .
        period: 60000
      steps:
        - setBody:
            id: setBody-d69b
            expression:
              simple:
                id: simple-645a
                expression: |-
                  <user>
                      <name>John</name>
                      <surname>Smith</surname>
                  </user>
        - unmarshal:
            id: unmarshal-e908
            jacksonXml:
              id: jacksonXml-ce0f
              unmarshalType: User
        - log:
            id: log-98d7
            message: ${body}

```
An error appears at startup:
```
org.apache.camel.FailedToCreateRouteException: Failed to create route route-6cc2 at: >>> Unmarshal[org.apache.camel.model.dataformat.JacksonXMLDataFormat@59c43561] <<< in route: Route(route-6cc2)[From[kamelet:timer-source?message=.&period... because of Error binding property (unmarshalType=User) with name: unmarshalType on bean: org.apache.camel.component.jacksonxml.JacksonXMLDataFormat@634f58d2 with value: User
...
Caused by: org.apache.camel.PropertyBindingException: Error binding property (unmarshalType=User) with name: unmarshalType on bean: org.apache.camel.component.jacksonxml.JacksonXMLDataFormat@634f58d2 with value: User
...
Caused by: org.apache.camel.RuntimeCamelException: org.apache.camel.NoTypeConversionAvailableException: No type converter available to convert from type: java.lang.String to the required type: java.lang.Class
...
Caused by: org.apache.camel.NoTypeConversionAvailableException: No type converter available to convert from type: java.lang.String to the required type: java.lang.Class
```
If I understood correctly, then the name of the class is perceived as a string. I tried writing: User, User.class and User.java .
What am I doing wrong?

**Comment by davsclaus** [comment, marked as answer]
The class name is FQN, so you should possible add the package name for the class, `om.dhls.camel.User` if you use that. Also make sure you have the user .java source together with the route yaml file when you run

**Reply by Bzelga** [reply]
Wow, thanks, it works, I didn't think to do that!
And if you don't use "package"?
After reading the answer, paragraph 4 says that the caravan does not use "package", so you can not do this? Then how do I write the class name to the unmarshal parameter?

**Reply by davsclaus** [reply]
Using java package for .java sources was not supported in the past. But newer version of Camel can now do that.

**Reply by davsclaus** [reply]
Btw where in the doc did you see that 4 paragraph?

**Reply by Bzelga** [reply]
There may be an inaccuracy in my message, because I use a translator :)
I was referring to the 4th paragraph of the message in the reply that I wrote at the very beginning.
